vendor/views/vendor_admin.py

```diff
@@ -393,36 +400,72 @@ class PaymentProcessorSiteSelectFormView(FormView):
     form_class = PaymentProcessorSiteSelectForm
 
     def get_success_url(self):
-        return reverse('vendor_admin:vendor-processor')
+        return reverse('vendor_admin:processor-lists')
 
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(*args, **kwargs)
-        processor_config = PaymentProcessorSiteSelectSiteConfig(Site.objects.get(pk=self.kwargs.get('pk')))
+
+        if self.kwargs.get('domain'):
+            site = Site.objects.get(domain=self.kwargs.get('domain'))
+            processor_config = PaymentProcessorSiteSelectSiteConfig(site)
+        else:
+            processor_config = PaymentProcessorSiteSelectSiteConfig()
         context['form'] = processor_config.get_form()
+
         return context
 
     def form_valid(self, form):
         site = Site.objects.get(pk=form.cleaned_data['site'])
         processor_config = PaymentProcessorSiteSelectSiteConfig(site)
         processor_config.save(form.cleaned_data["payment_processor"], "payment_processor")
+        
         return redirect('vendor_admin:vendor-processor-lists')
 
 
 class AuthorizeNetIntegrationView(FormView):
-    template_name = "vendor/authorizenet_integration.html"
+    template_name = "vendor/integration_form.html"
     form_class = AuthorizeNetIntegrationForm
     success_url = reverse_lazy('authorizenet-integration')
 
     def get_context_data(self, *args, **kwargs):
         context = super().get_context_data(**kwargs)
         authorizenet_integration = AuthorizeNetIntegration(get_site_from_request(self.request))
+        context['integration_name'] = _("AuthorizeNet Integration")
+
         if authorizenet_integration.instance:
             context['form'] = AuthorizeNetIntegrationForm(instance=authorizenet_integration.instance)
         else:
             context['form'] = AuthorizeNetIntegrationForm()
+
         return context
     
     def form_valid(self, form):
         authorizenet_integration = AuthorizeNetIntegration(get_site_from_request(self.request))
         authorizenet_integration.save(form.cleaned_data)
+
         return super().form_valid(form)
+
+
+class StripeIntegrationView(FormView):
+    template_name = "vendor/integration_form.html"
+    form_class = StripeIntegrationForm
+    success_url = reverse_lazy('stripe-integration')
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(**kwargs)
+        stripe_integration = StripeIntegration(get_site_from_request(self.request))
+        context['integration_name'] = _("Stripe Integration")
+
+        if stripe_integration.instance:
+            context['form'] = StripeIntegrationForm(instance=stripe_integration.instance)
```

Comment:
you might be able to just do `context['form'] = StripeIntegrationForm(instance=stripe_integration.instance)` without an if statement if stripe_integration.instance returns None when not set

Reply:
You are right thank you!!!!